src/Evolution/DgSubcell/Actions/ReconstructionCommunication.hpp

```diff
@@ -0,0 +1,411 @@
+
+  template <typename DbTags, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename ParallelComponent,
+            typename Metavariables>
+  static std::tuple<db::DataBox<DbTags>&&> apply(
+      db::DataBox<DbTags>& box, tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    using variables_tag = typename Metavariables::system::variables_tag;
+
+    ASSERT(db::get<Tags::ActiveGrid>(box) == ActiveGrid::Subcell,
+           "The SendDataForReconstruction action can only be called when "
+           "Subcell is the active scheme.");
+
+    db::mutate<Tags::NeighborDataForReconstructionAndRdmpTci<Dim>>(
+        make_not_null(&box),
+        [](const auto neighbor_data_ptr, const auto& active_vars) noexcept {
+          auto [max_of_vars, min_of_vars] =
+              rdmp_max_min(active_vars, {}, false);
+
+          // Clear the previous neighbor data and add current local data
+          neighbor_data_ptr->clear();
+          (*neighbor_data_ptr)[std::pair{
+              Direction<Dim>::lower_xi(),
+              ElementId<Dim>::external_boundary_id()}] =
+              NeighborData{{}, std::move(max_of_vars), std::move(min_of_vars)};
+        },
+        db::get<variables_tag>(box));
+
+    const Mesh<Dim>& dg_mesh = db::get<::domain::Tags::Mesh<Dim>>(box);
+    const Mesh<Dim>& subcell_mesh = db::get<Tags::Mesh<Dim>>(box);
+    const Element<Dim>& element = db::get<::domain::Tags::Element<Dim>>(box);
+    const size_t ghost_zone_size =
+        Metavariables::SubcellOptions::ghost_zone_size(box);
+    // Note: we currently always send the conserved variables, but will need to
+    // generalize that to allow the user to choose what to send. For example,
+    // with relativistic hydrodynamics it's desirable to reconstruct
+    //    u_i = u^0(lapse * v_i - shift_i)
+    // since this guarantees that the Lorentz factor is physical after
+    // reconstruction.
+    DirectionMap<Dim, bool> directions_to_slice{};
+    for (const auto& direction_neighbors : element.neighbors()) {
+      if (direction_neighbors.second.size() == 0) {
+        directions_to_slice[direction_neighbors.first] = false;
+      } else {
+        directions_to_slice[direction_neighbors.first] = true;
+      }
+    }
+    // Optimization note: could save a copy+allocation if we moved
+    // all_sliced_data when possible before sending.
+    const DirectionMap<Dim, std::vector<double>> all_sliced_data = slice_data(
+        db::mutate_apply(GhostDataMutator{}, make_not_null(&box)),
+        subcell_mesh.extents(), ghost_zone_size, directions_to_slice);
+
+    auto& receiver_proxy =
+        Parallel::get_parallel_component<ParallelComponent>(cache);
+    const NeighborData& local_neighbor_data =
```

Comment:
hasn't this been called self_neighbor_data in other places?

Reply:
Maybe, I don't remember off the top of my head.